coding-guidelines/coding-guidelines.md

```diff
@@ -0,0 +1,243 @@
+# Coding Guidelines
+
+## Line length
+One Line in JavaScript should have a maximal length of 80 characters.
+
+## Indentation and Line-Breaks
+Rules:
+- Indentation is done with 2 spaces in most cases.
+- Line breaks are only done if needed, try to use the 80 character limit.
+- Strings are splitted at the latest possible position.
+  - At the latest whitespace character.
+  - The second line is indented with 2 spaces.
+- Object Literals are kept on one line if possible.
+  - If the line gets to long put each attribute on a single line.
+  - Multi level object literals should be splitted in to multiple lines. Each level in the object should have 2 spaces as indentation.
+
+Examples:
+```JS
+switch (a) {
+  case "a":
+    break;
+  case "b":
+    break;
+}
+
+var a,
+  b,
+  c;
+
+function test() {
+  var a,
+    b,
+    c;
+}
+
+test(
+  "A Very long string which is longer then one line, which is 80 characters " +
+    "in this case. This demonstrates how the second line is indented",
+  "Another string parameter.",
+  {and: "a Literal Object", as: "a parameter", type: "short"},
+  {
+    yet: "another long parameter Object",
+    which: "needs to be split in multiple",
+    lines: 3
+  }
+);
+
+test({
+  a: {
+    b: 10
+  },
+  c: 20,
+  d: 30,
+  e: 40,
+  f: 50,
+  g: 60,
+  h: {
+    i: 70
+  }
+});
+```
+
+## Whitespace
+Rules:
+- No Trailing spaces should be used at the end of a line.
+- No multiple Blank lines.
+
+## Functions
+Rules:
+  - Function blocks should have a space before the definition.
+  - Function should have a space before the opening parentheses.
+  - Named Functions should not have a space before the opening function parentheses.
+  - Functions can be on a single line.
+  - Anonymous Functions need to have a space before the opening function parentheses.
+  - Arrow functions do have the same constraints
+
+Examples:
+```JS
+function test() {
+  return;
+}
+
+function test() { var a = 10; return a; }
+
+var test = function () {var b = 10; return b;}
+
+var es6 = parameter => parameter();
```

Comment:
Maybe some explanation?
"If the arrow functions fits on one line, we don't use curly braces." Or such..